### PR TITLE
Clarify support for generated code

### LINKED ISSUE #3515: Register generated files via .gitattributes?
We use a lot of code generation in our Elm project and the generated files can make it hard to read pull requests. They aren't documentation and `vendored` doesn't make sense here either. Is it possible to register generated files via .gitattributes? If not, would a pull request for this be welcomed?

### PR DESCRIPTION
**Pull request by jelder**
Addresses #3515.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,13 +32,15 @@ Linguist supports a number of different custom overrides strategies for language
...
 
-Add a `.gitattributes` file to your project and use standard git-style path matchers for the files you want to override to set `linguist-documentation`, `linguist-language`, and `linguist-vendored`. `.gitattributes` will be used to determine language statistics and will be used to syntax highlight files. You can also manually set syntax highlighting using [Vim or Emacs modelines](#using-emacs-or-vim-modelines).
+Add a `.gitattributes` file to your project and use standard git-style path matchers for the files you want to override to set `linguist-documentation`, `linguist-language`, `linguist-vendored`, and `linguist-generated`. `.gitattributes` will be used to determine language statistics and will be used to syntax highlight files. You can also manually set syntax highlighting using [Vim or Emacs modelines](#using-emacs-or-vim-modelines).
 
 ```
 $ cat .gitattributes
 *.rb linguist-language=Java
 ```
 
+#### Vendored Code
```
Comment (Alhadis) [inline README.md]: I'd lowercase the letter `C` here so the casing stays consistent with the other subheadings.

File: README.md [hunk 2]
```diff
@@ -59,15 +63,14 @@ project-docs/* linguist-documentation
 docs/formatter.rb linguist-documentation=false
 ```
 
-#### Generated file detection
+#### Generated Code
```
Comment (Alhadis) [inline README.md]: Same here. Better maintain consistent capitalisation.

File: README.md [hunk 3]
```diff
@@ -59,15 +63,14 @@ project-docs/* linguist-documentation
...
 
 Not all plain text files are true source files. Generated files like minified js and compiled CoffeeScript can be detected and excluded from language stats. As an added bonus, unlike vendored and documentation files, these files are suppressed in diffs.
 
-```ruby
-Linguist::FileBlob.new("underscore.min.js").generated? # => true
 ```
-
-See [Linguist::Generated#generated?](https://github.com/github/linguist/blob/master/lib/linguist/generated.rb).
+$ cat .gitattributes
++**/Api.elm linguist-generated=true
```
Comment (Alhadis) [inline README.md]: Was the leading `+` intentional?

**Comment by Alhadis** [comment]
Looks good! Thanks!

Hey, just curious: why did you opt for a double-globstar for `**/Api.elm`? `Api.elm` does the same and is more reader-friendly.

**Comment by jelder** [comment]
I didn't realize `.gitattributes` matching worked that way. Updated.

**Comment by Alhadis** [comment]
LGTM. /cc @lildude

**Pull request merged**